File: Mobile Applications/android/app/src/main/java/com/example/activityplannerapp/activities/domain/Activity.kt
package com.example.activityplannerapp.activities.domain

import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.time.LocalDate
import java.time.LocalTime
import java.time.format.DateTimeFormatter

//data class Activity(
//    val id: Int,
//    var title: String,
//    var type: String,
//    var date: LocalDate,
//    var startTime: LocalTime,
//    var endTime: LocalTime,
//    var description: String,
//) {
//
//    private fun formatDate(date: LocalDate): String {
//        val formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy")
//        return date.format(formatter)
//    }
//
//    private fun formatTime(time: LocalTime): String {
//        val formatter = DateTimeFormatter.ofPattern("HH:mm")
//        return time.format(formatter)
//    }
//
//    override fun toString(): String {
//        return "$title \nDate: ${formatDate(date)} \nStart Time: ${formatTime(startTime)} \nEnd Time: ${formatTime(endTime)} \nDetails: $description"
//    }
//}

@Serializable
@Entity(tableName = "activities")
data class Activity(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    var title: String,
    var type: String,

    @Contextual
    @Serializable(with = LocalDateSerializer::class)
    var date: LocalDate,

    @Contextual
    @Serializable(with = LocalTimeSerializer::class)
    var startTime: LocalTime,

    @Contextual
    @Serializable(with = LocalTimeSerializer::class)
    var endTime: LocalTime,
    var description: String,
) {
    private fun formatDate(date: LocalDate): String {
        val formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy")
        return date.format(formatter)
    }

    private fun formatTime(time: LocalTime): String {
        val formatter = DateTimeFormatter.ofPattern("HH:mm")
        return time.format(formatter)
    }

    override fun toString(): String {
        return "$title \nDate: ${formatDate(date)} \nStart Time: ${formatTime(startTime)} \nEnd Time: ${formatTime(endTime)} \nDetails: $description"
    }
}

File: Mobile Applications/flutter_app/lib/Domain/Activity.dart


import 'package:flutter/material.dart';
import 'package:intl/intl.dart';

class Activity {
  static int currentId = 0;
  late int id;
  String title;
  String type;
  DateTime date;
  TimeOfDay startTime;
  TimeOfDay endTime;
  String description;

  Activity(this.title, this.type, this.date, this.startTime, this.endTime, this.description) {
    id = ++currentId;
  }

  Activity.namedConstructor(this.id, this.title, this.type, this.date, this.startTime, this.endTime, this.description);

  String formatDate(DateTime date) {
    return DateFormat('dd/MM/yyyy').format(date);
  }

  String formatTime(TimeOfDay time) {
    final now = DateTime.now();
    final dateTime = DateTime(now.year, now.month, now.day, time.hour, time.minute);
    final formatter = DateFormat('HH:mm');
    return formatter.format(dateTime);
  }

  @override
  String toString() {
    return "$title\nDate: ${formatDate(date)}\nStart Time: ${formatTime(startTime)}\nEndTime: ${formatTime(endTime)}\nDetails: $description";
  }
}